src/main/java/com/hotels/housekeeping/service/impl/FileSystemHousekeepingService.java

```diff
@@ -47,39 +48,62 @@ public FileSystemHousekeepingService(LegacyReplicaPathRepository legacyReplicaPa
     LOG.warn("{}.fixIncompleteRecord(LegacyReplicaPath) should be removed in future.", getClass());
   }
 
+  private FileSystem fileSystemForPath(Path path) {
+    try {
+      return path.getFileSystem(conf);
+    } catch (IOException e) {
+      throw new HousekeepingException(e);
+    }
+  }
+
+  private void housekeepPath(LegacyReplicaPath cleanUpPath) {
+    final Path path = new Path(cleanUpPath.getPath());
+    final Path rootPath;
+    final FileSystem fs;
+    try {
+      fs = fileSystemForPath(new Path(cleanUpPath.getPath()));
+      LOG.info("Attempting to delete path '{}' from file system", cleanUpPath);
+      if (fs.exists(path)) {
+        fs.delete(path, true);
+        LOG.info("Path '{}' has been deleted from file system", cleanUpPath);
+      } else {
+        LOG.warn("Path '{}' does not exist.", cleanUpPath);
+      }
+      rootPath = deleteParents(fs, path, cleanUpPath.getPathEventId());
+    } catch (Exception e) {
+      LOG.warn("Unable to delete path '{}' from file system. Will try next time. {}", cleanUpPath, e.getMessage());
+      return;
+    }
+
+    try {
+      if (oneOfMySiblingsWillTakeCareOfMyAncestors(path, rootPath, fs) || thereIsNothingMoreToDelete(fs, rootPath)) {
+        // BEWARE the eventual consistency of your blobstore!
+        try {
+          LOG.info("Deleting path '{}' from housekeeping database", cleanUpPath);
+          legacyReplicaPathRepository.delete(cleanUpPath);
+        } catch (ObjectOptimisticLockingFailureException e) {
+          LOG.debug(
+              "Failed to delete path '{}': probably already cleaned up by process running at same time. Ok to ignore. {}",
+              cleanUpPath, e.getMessage());
+        }
+      }
+    } catch (Exception e) {
+      LOG.warn("Path '{}' was not deleted from the housekeeping database. {}", cleanUpPath, e.getMessage());
```

Comment:
This does prevent housekeeping from failing but it won't delete the path from the repo, so we'll get this warning on every run of housekeeping or am I missing something?

Reply:
This block is only reached if `legacyReplicaPathRepository.delete(cleanUpPath);` fails, signifying a failure to delete the specified path from the Housekeeping database

Reply:
if `oneOfMySiblingsWillTakeCareOfMyAncestors(path, rootPath, fs) || thereIsNothingMoreToDelete(fs, rootPath)` throws an exception which I thought was why we got the error in the first place, we'll hit the catch

Reply:
Ah right, I was looking at wrong try block.. I'm hesitant to delete data from the Housekeeping database here as it may fail for another unforeseen reason that isn't the path being deleted already, do you think we should just do the deletion anyway?

Reply:
I guess we need to know exactly at what point we got the error and check what path we were trying to delete is that some parent path or something else? What would happen if we warn and run again, failure at same place?

Reply:
This becomes complex cause of eventual consistency and the likes, I would rather leave it like this, they will get a small warning in their logs when they run their application but I would rather that over leaving behind data that I may be paying for